### PR TITLE
add create review proposition functionality, fix order create

### PR DESCRIPTION
**Pull request by olegbondur**
Task 119: Create on backend controller and service for save review proposition
[TASK-119]

https://car-service.visualstudio.com/CarService/_workitems/edit/119

### PR Author Checklist
- [x] Ticket created and linked in the PR description
- [x] Changes are followed [code conventions](https://wiki/Code+Conventions) 
- [x] Changes are related to a single context
- [ ] Changes are dev tested
- [ ] Unit tests written and passing locally
- [ ] PR link added to еру ticket
- [x] Conflicts resolved

#### For Bug Fixes
- [ ] Regression Test written
- [ ] Logging added

### Reviewer Checklist
- [ ] Tested locally
- [ ] Validated against JIRA description
- [ ] Verified author checklist

* See ["Code review process"](https://wiki/Code+Review+Process) wiki for more details

### CODE REVIEW

File: CarService.Api/Controllers/OrdersController.cs [hunk 1]
```diff
@@ -37,6 +37,25 @@ public OrdersController(IOrderService orderService)
...
+        /// <remarks>TODO detail description</remarks>
+        /// <param name="reviewPropositionDto">Orders' model</param> 
+        /// <response code="200">review proposition create successful</response>
+        /// <response code="401">Unauthorized user</response> 
+        /// <response code="500">Internal Server Error</response> 
+        [Authorize]
+        [HttpPost("create-review-proposition")]
+        public async Task<IActionResult> CreateReviewProposition([FromBody]ReviewPropositionCreationDto reviewPropositionDto)
+        {
+            string email = User.Identity.Name;
```
Comment (AndrewGumenyuk) [inline CarService.Api/Controllers/OrdersController.cs]: Isn't enough the authorize attribute?

File: CarService.Api/Services/OrderService.cs [hunk 2]
```diff
@@ -63,6 +66,28 @@ public OrderService(UserManager<User> userManager, IUnitOfWorkFactory unitOfWork
             }
         }
 
+        public async Task CreateReviewProposition(string email, ReviewPropositionCreationDto reviewPropositionDto)
+        {
+            var user = await _userManager.FindByEmailAsync(email);
+
+            if (user != null)
```
Comment (AndrewGumenyuk) [inline CarService.Api/Services/OrderService.cs]: You've already checked it inside the Controller, isn't enough it?

File: CarService.Api/Services/OrderService.cs [hunk 3]
```diff
@@ -63,6 +66,28 @@ public OrderService(UserManager<User> userManager, IUnitOfWorkFactory unitOfWork
...
+            {
+                using (IUnitOfWork unitOfWork = _unitOfWorkFactory.Create())
+                {
+                    IRepository<ReviewProposition> reviewProposition = unitOfWork.Repository<ReviewProposition>();
+                    reviewProposition.Add(new ReviewProposition
+                    {
+                        MechanicId = user.Id,
+                        OrderId = reviewPropositionDto.OrderId,
+                        Comment = reviewPropositionDto.ReviewDescription,
```
Comment (AndrewGumenyuk) [inline CarService.Api/Services/OrderService.cs]: The comment doesn't equal to Description

File: CarService.Api/Services/OrderService.cs [hunk 4]
```diff
@@ -63,6 +66,28 @@ public OrderService(UserManager<User> userManager, IUnitOfWorkFactory unitOfWork
...
 
+        public async Task CreateReviewProposition(string email, ReviewPropositionCreationDto reviewPropositionDto)
+        {
+            var user = await _userManager.FindByEmailAsync(email);
+
+            if (user != null)
+            {
+                using (IUnitOfWork unitOfWork = _unitOfWorkFactory.Create())
+                {
+                    IRepository<ReviewProposition> reviewProposition = unitOfWork.Repository<ReviewProposition>();
```
Comment (AndrewGumenyuk) [inline CarService.Api/Services/OrderService.cs]: Wow, it looks very strange

File: CarService.Api/Controllers/OrdersController.cs [hunk 5]
```diff
@@ -37,6 +37,25 @@ public OrdersController(IOrderService orderService)
...
+        /// <summary>Create mechanic review proposition</summary>
+        /// <returns>an IActionResult</returns>
+        /// <remarks>TODO detail description</remarks>
+        /// <param name="reviewPropositionDto">Orders' model</param> 
+        /// <response code="200">review proposition create successful</response>
+        /// <response code="401">Unauthorized user</response> 
+        /// <response code="500">Internal Server Error</response> 
+        [Authorize]
+        [HttpPost("create-review-proposition")]
+        public async Task<IActionResult> CreateReviewProposition([FromBody]ReviewPropositionCreationDto reviewPropositionDto)
```
Comment (AndrewGumenyuk) [inline CarService.Api/Controllers/OrdersController.cs]: Are you going to validate this model into your controller or somewhere else

File: CarService.Api/Controllers/OrdersController.cs (hunk 1 above)
Comment (olegbondur) [reply CarService.Api/Controllers/OrdersController.cs]: No, it's not necessary. will be deleted in the following commit

File: CarService.Api/Services/OrderService.cs (hunk 2 above)
Comment (olegbondur) [reply CarService.Api/Services/OrderService.cs]: if(user!=null) will be deleted from all Services' method in next commit

File: CarService.Api/Services/OrderService.cs (hunk 3 above)
Comment (olegbondur) [reply CarService.Api/Services/OrderService.cs]: reviewPropositionDto.ReviewDescription will be changed to reviewPropositionDto.ReviewComment in next commit

File: CarService.Api/Controllers/OrdersController.cs (hunk 5 above)
Comment (olegbondur) [reply CarService.Api/Controllers/OrdersController.cs]: Validation will be added on backend side in next commit

File: CarService.Api/Services/OrderService.cs (hunk 4 above)
Comment (olegbondur) [reply CarService.Api/Services/OrderService.cs]: Could you explain what you mean? Maybe we must change IRepository<ReviewProposition> reviewProposition = unitOfWork.Repository<ReviewProposition>(); to var = unitOfWork.Repository<ReviewProposition>(); ?

Thanks for review

**Review by LytvyniukDima — APPROVED** [review]

**Pull request merged**